models.py

```diff
@@ -930,3 +930,21 @@ class AdminMessage(Base):
         index=True,
     )
     archived_reason = Column(String)
+
+
+class TaskQueue(Base):
+    # start without indexing since the idea is to remove on calculation start
+    # only meant as persistent layer, queue itself accesses cache
+    __tablename__ = Tablenames.TASK_QUEUE.value
+    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
+    project_id = Column(
+        UUID(as_uuid=True),
+        ForeignKey(f"{Tablenames.PROJECT.value}.id", ondelete="CASCADE"),
+    )
+    type = Column(String)  # enum.TaskType e.g. EMBEDDING
```

Comment:
I would rename the column to `task_type` to differ from built-in `type`

Reply:
we use type quite often as column name (e.g. information sources) -> still a valid point so I changed it